### PR TITLE
rough draft of a tracing blog post

### PR DESCRIPTION
**Pull request by hawkw**
This is a rough draft of a blog post introducing the tracing crate. It would be great to get some feedback prior to publishing.

Signed-off-by: Eliza Weisman <eliza@buoyant.io>

### CODE REVIEW

File: content/blog/2019-08-tracing.md [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+`tracing` is more than a logging library: it implements scoped, contextual, and
+structured diagnostic instrumentation. This allows users to trace logical
+contexts in the application through time, even as the actual flow of execution moves
+between those contexts.
+
+### Spans
+
+To record the flow of execution, `tracing` introduces the concept of _spans_. Unlike a log
+line that represents a moment in time, a span models a period of time with a
+beginning and an end. When a a program begins executing in a context or performing
```
Comment (yaahc) [inline content/blog/2019-08-tracing.md]: a a

File: content/blog/2019-08-tracing.md [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+
+Contributions in all these areas will be welcomed eagerly. We're all looking
+forward to seeing what the community will build on top of the platform
+that `tracing` provides!
+
+[linkerd]: https://github.com/linkerd/linkerd2-proxy
+[vector]: https://github.com/timberio/vector
+[OpenTelemetry]: https://opentelemetry.io/
+[Jaeger]: https://www.jaegertracing.io/
+[core-crates]: https://crates.io/crates/tracing-crates
```
Comment (tobz) [inline content/blog/2019-08-tracing.md]: Bad link.

File: content/blog/2019-08-tracing.md [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+
+```rust
+use tracing::trace;
+
+let bytes_read = ...;
+let num_processed = ...;
+
+// ...
+
+trace!(bytes_read, messages = num_processed);
```
Comment (yaahc) [inline content/blog/2019-08-tracing.md]: worried the lack of a message field in this example could be confusing to readers who try to add a message or use structured data in a macro call that already has a message.

File: content/blog/2019-08-tracing.md [hunk 4]
```diff
@@ -0,0 +1,259 @@
...
+
+* Integrating with distributed tracing systems such as [OpenTelemetry] or
+  [Jaeger].
+* Building out richer instrumentation in the Tokio runtime.
+* Integration with more libraries and frameworks.
+* Writing `Subscriber`s to implement more ways of collecting trace data, such as
+  metrics, profiling, et cetera.
+* Helping to stabilize experimental crates.
+
+Contributions in all these areas will be welcomed eagerly. We're all looking
```
Comment (yaahc) [inline content/blog/2019-08-tracing.md]: This might be a good location to add a link to the gitter.

**Comment by hawkw** [comment]
@carllerche I definitely agree. I wasn't sure how to translate the demo, which is kind of inherently  interactive, into text. Any suggestions?

**Comment by carllerche** [comment]
@hawkw You don't have to go into details of the demo, you can link to the full repo for those who want to dig in.

Roughly, in the blog post, you can do:

- Imagine a web app that is concurrently receiving requests... there is a problem.
- Here is the log output w/o tracing
- Here it is w/ tracing
- Now we filter by requests to the buggy endpoing
   - Trace data for child events are included.
   - And look, the source of the problem becomes obvious.

**Comment by hawkw** [comment]
@carllerche I've updated this to include a quick walk through the demo.

File: content/blog/2019-08-tracing.md [hunk 5]
```diff
@@ -0,0 +1,485 @@
...
+For example, we might also consume the above data by counting the number of
+requests recieved for different paths or HTTP methods. By looking at the
+structure of the span tree as well as at key-value data, we can even do things
+like recording the entire lifespan of a request only when it ended with an
+error.
+
+## A Worked Example
+
+To demonstrate the value of this kind of diagnostics, let's take a look at an
+example. In this example, we've written a small HTTP service, using [`tower`],
```
Comment (davidbarsky) [inline content/blog/2019-08-tracing.md]: I think you can drop the commas on either side of `, using [tower],`.

**Pull request merged**